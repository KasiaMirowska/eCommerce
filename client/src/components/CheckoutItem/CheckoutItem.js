import React from 'react';
import './CheckoutItem.styles.scss';
import { connect } from 'react-redux';
import { addItemToCart, clearItemFromCart, removeItem } from '../../redux/cart/cart.actions';

const mapDispatchToProps = dispatch => ({
    clearItemFromCart: item => dispatch(clearItemFromCart(item)),
    addItem: item => dispatch(addItemToCart(item)),
    removeItem: item => dispatch(removeItem(item))
});

export const CheckoutItem = (props) => {
    
    const { imageUrl, name, quantity, price, clearItemFromCart, addItem, removeItem } = props;

    //as props we need to pass cartItems but also the item itself in order to be able to add or remove another items to cart
    return (
        <div className='checkout-item'>
            <div className='image-container'>
                <img src={imageUrl} alt='checkout item' />
            </div>
            <span className='name'>{name}</span>

            <span className='quantity'>
                <div className='arrow' id='decrease' onClick={()=> removeItem(props)}>&#10094;</div>
                <span className='value'>{quantity}</span>
                <div className='arrow' id='increase' onClick={()=> addItem(props)}>&#10095;
                </div>
            </span>
            <span className='price'>{price}</span>
            <div className='remove-button' onClick={() => clearItemFromCart(props)}>&#10005;</div>
        </div>
    )
}

export default connect(null, mapDispatchToProps)(CheckoutItem);