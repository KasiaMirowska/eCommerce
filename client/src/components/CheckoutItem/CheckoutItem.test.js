import React from 'react';
import {shallow} from 'enzyme';
import {CheckoutItem} from '../../components/CheckoutItem/CheckoutItem';


describe('CheckoutItem', () => {
    const mockAddItem = jest.fn();
    const mockRemoveItem = jest.fn();
    const mockClearItem = jest.fn();

    const props = {
        imageUrl: 'url', 
        name: 'brown hat', 
        quantity: 1, 
        price: 24, 
        clearItemFromCart: mockClearItem , 
        addItem: mockAddItem, 
        removeItem: mockRemoveItem }

    const wrapper = shallow(<CheckoutItem {...props} />);

    it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('on click increases or decreases quantity', () => {
        let increase = wrapper.find('#increase');
        increase.simulate('click');
        expect(mockAddItem).toHaveBeenCalled();
        expect(props.quantity).toBe(+1);

        let decrease = wrapper.find('#decrease');
        decrease.simulate('click');
        expect(mockRemoveItem).toHaveBeenCalled();
        expect(props.quantity).toBe(1);
    });

    it('clears the cart on click', () => {
        let cleared = wrapper.find('.remove-button');
        cleared.simulate('click');
        expect(mockClearItem).toHaveBeenCalled();
    })

})
