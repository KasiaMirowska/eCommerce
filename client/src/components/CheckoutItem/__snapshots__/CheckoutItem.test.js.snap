// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem renders as expected 1`] = `ShallowWrapper {}`;
